bundles/org.openhab.binding.visualcrossing/src/main/java/org/openhab/binding/visualcrossing/internal/api/VisualCrossingApi.java

```diff
@@ -0,0 +1,177 @@
+     *            February 2021. You can also request the information for a specific time for a single date by including
+     *            time into the date1 field using the format yyyy-MM-ddTHH:mm:ss. For example 2020-10-19T13:00:00.The
+     *            results are returned in the ‘currentConditions’ field and are truncated to the hour requested (i.e.
+     *            2020-10-19T13:59:00 will return data at 2020-10-19T13:00:00).
+     * @param dateTo (optional) Is the end date for which to retrieve weather data. This value may only be used when a
+     *            date1 value is given. When both date1 and date2 values are given, the query is inclusive of date2 and
+     *            the weather data request period will end on midnight of the date2 value. All dates and times are in
+     *            local time of the specified location and should be in the format yyyy-MM-dd.
+     * @return WeatherResponse from server
+     * @throws VisualCrossingApiException in case of an error
+     * @throws VisualCrossingAuthException if the API key is not correct
+     * @throws VisualCrossingRateException if the rate limit was exceeded
+     * @see <a href="https://www.visualcrossing.com/resources/documentation/weather-api/timeline-weather-api/">Timeline
+     *      Weather API</a>
+     */
+    public WeatherResponse timeline(String location, @Nullable UnitGroup unitGroup, @Nullable String lang,
+            @Nullable String dateFrom, @Nullable String dateTo)
+            throws VisualCrossingApiException, VisualCrossingAuthException, VisualCrossingRateException {
+        if (unitGroup == null) {
+            unitGroup = METRIC;
+        }
+        if (dateFrom == null && dateTo != null) {
+            throw new VisualCrossingApiException("When passing dateTo you need to pass dateFrom!");
+        }
+
+        var escapedLocation = URLEncoder.encode(location, UTF_8).replaceAll("\\+", "%20");
+
+        var url = new StringBuilder(baseUrl)//
+                .append("/VisualCrossingWebServices/rest/services/timeline/")//
+                .append(escapedLocation);
+        if (dateFrom != null) {
+            url.append("/").append(dateFrom);
+            if (dateTo != null) {
+                url.append("/").append(dateTo);
+            }
+        }
+        url.append("?key=").append(apiKey)//
+                .append("&contentType=json")//
+                .append("&unitGroup=").append(unitGroup.unit);
+        if (lang != null && !lang.isEmpty()) {
+            url.append("&lang=").append(lang);
+        }
+        var response = restClient.get(url.toString());
+
+        try {
+            var weatherResponse = requireNonNull(gson.fromJson(response, WeatherResponse.class));
+            addCost(weatherResponse);
+            return weatherResponse;
+        } catch (JsonSyntaxException e) {
+            var target = WeatherResponse.class.getSimpleName();
+            logger.debug("Cannot parse {} from JSON:\n{}", target, response, e);
```

Comment:
When the json contents (or other verbose request/reponse/data) are logged, it should go to trace. 
You could make a condiutional mix of both.